src/L1FarmProxy.sol

```diff
@@ -0,0 +1,102 @@
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    ) external;
+}
+
+contract L1FarmProxy {
+    mapping (address => uint256) public wards;
+    uint32  public minGasLimit;
+    uint224 public rewardThreshold; // TODO: should we still have it? we don't have to anymore since this contract doesn't hold eth to waste. If not then we rely on dss-cron's defined intervals for timing - https://github.com/makerdao/dss-cron/blob/master/src/VestRewardsDistributionJob.sol#L141. Opting for now to leave it to maintain symmetry with Arbitrum and with the L2 proxy.
+
+    address public immutable rewardsToken;
+    address public immutable remoteToken;
+    address public immutable l2Proxy;
+    L1TokenBridgeLike public immutable l1Bridge;
+
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event RewardAdded(uint256 reward);
+
+    constructor(address _rewardsToken, address _remoteToken, address _l2Proxy, address _l1Bridge) {
+        rewardsToken = _rewardsToken;
+        remoteToken  = _remoteToken;
+        l2Proxy      = _l2Proxy;
+        l1Bridge     = L1TokenBridgeLike(_l1Bridge);
+
+        GemLike(_rewardsToken).approve(_l1Bridge, type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "L1FarmProxy/not-authorized");
+        _;
+    }
+
+    function rely(address usr) external auth { wards[usr] = 1; emit Rely(usr); }
+    function deny(address usr) external auth { wards[usr] = 0; emit Deny(usr); }
+
+    // @notice Validation of the `data` boundaries is outside the scope of this 
+    // contract and is assumed to be carried out in the corresponding spell process
+    function file(bytes32 what, uint256 data) external auth {
+        if      (what == "minGasLimit")     minGasLimit     = uint32(data);
+        else if (what == "rewardThreshold") rewardThreshold = uint224(data);
+        else revert("L1FarmProxy/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // @notice Allow governance to recover potentially stuck tokens
+    function recover(address token, address receiver, uint256 amount) external auth {
+        GemLike(token).transfer(receiver, amount);
+    }
+
+    // @notice Send reward to L2 farm proxy
+    function notifyRewardAmount(uint256 reward) external {
+        (uint32 minGasLimit_, uint256 rewardThreshold_) = (minGasLimit, rewardThreshold);
+        require(reward > rewardThreshold_, "L1FarmProxy/reward-too-small");
+
+        l1Bridge.bridgeERC20To({
+            _localToken:   rewardsToken,
+            _remoteToken:  remoteToken,
+            _to :          l2Proxy,
+            _amount :      reward,
+            _minGasLimit:  minGasLimit_,
+            _extraData :   bytes("")
+        });
+
+        emit RewardAdded(reward);
+    }
+}
+
```

Comment:
There are 2 spaces.

Reply:
Handled.

Reply:
Inlined (also exists in `arbitrum-farms`).